python/lancedb/query.py

```diff
@@ -260,20 +260,30 @@ def to_pydantic(self, model: Type[LanceModel]) -> List[LanceModel]:
             for row in self.to_arrow().to_pylist()
         ]
 
-    def limit(self, limit: int) -> LanceQueryBuilder:
+    def limit(self, limit: Union[int, None]) -> LanceQueryBuilder:
         """Set the maximum number of results to return.
 
         Parameters
         ----------
         limit: int
             The maximum number of results to return.
+            By default the query is limited to the first 10.
+            Call this method and pass in any negative value
+            or None to remove the limit.
+            *WARNING* if you have a large dataset, removing
+            the limit can potentially result in reading a
+            large amount of data into memory and cause
+            out of memory issues.
```

Comment:
Even if I call `to_batches`?

Reply:
we don't have `to_batches` api yet. I'll create a follow-up issue